src/GraphArray.chpl

```diff
@@ -0,0 +1,460 @@
+    var v_weight: borrowed SymEntry(int);
+
+    /*    The weitht of every edge in the graph, name */
+    var e_weightName : string;
+
+    /*    The weitht of every edge in the graph, array value */
+    var e_weight : borrowed SymEntry(int);
+
+
+
+    /* 
+     * The following version we will init differnt kind of arrays
+     * this is for src, dst, start_i, neighbour, v_weight and e_weight arrays
+     */
+    proc init( numv:int, nume:int, dire:bool, srcNameA: string, dstNameA: string, 
+               startNameA:string,neiNameA: string, vweiNameA: string, 
+               eweiNameA:string, st: borrowed SymTab) {
+      n_vertices=numv;
+      n_edges=nume;
+      directed=dire;
+      
+
+      srcName = srcNameA;
+      // The try! is needed here because init cannot throw
+      var gs = try! st.lookup(srcName);
+      var tmpsrc = toSymEntry(gs, int): unmanaged SymEntry(int);
+      src=tmpsrc;
+
+      dstName = dstNameA;
+      // The try! is needed here because init cannot throw
+      var ds = try! st.lookup(dstName);
+      var tmpdst = toSymEntry(ds, int): unmanaged SymEntry(int);
+      dst=tmpdst;
+
+      startName = startNameA;
+      // The try! is needed here because init cannot throw
+      var starts = try! st.lookup(startName);
+      var tmpstart_i = toSymEntry(starts, int): unmanaged SymEntry(int);
+      start_i=tmpstart_i;
+
+      neighbourName = neiNameA;
+      // The try! is needed here because init cannot throw
+      var neis = try! st.lookup(neighbourName);
+      var tmpneighbour = toSymEntry(neis, int): unmanaged SymEntry(int);
+      neighbour=tmpneighbour;
+
+      v_weightName = vweiNameA;
+      // The try! is needed here because init cannot throw
+      var vweis = try! st.lookup(v_weightName);
+      // I want this to be borrowed, but that throws a lifetime error
+      var tmpv_weight = toSymEntry(vweis, int): unmanaged SymEntry(int);
+      v_weight=tmpv_weight;
+
+      e_weightName = eweiNameA;
+      // The try! is needed here because init cannot throw
+      var eweis = try! st.lookup(e_weightName);
+      var tmpe_weight = toSymEntry(eweis, int): unmanaged SymEntry(int);
+      e_weight=tmpe_weight;
+
+    }
+
+
+  } // class SegGraph
+
+
+
+
+  /**
+   * We use several arrays and intgers to represent a basic directed graph 
+   * Instances are ephemeral, not stored in the symbol table. Instead, attributes
+   * of this class refer to symbol table entries that persist. This class is a
+   * convenience for bundling those persistent objects and defining graph-relevant
+   * operations.
+   * Now we  copy from SegSArray, we need change more in the future to fit a graph
+   */
+  class SegGraphD {
+ 
+    /*    The starting indices for each string*/
+    var n_vertices : int;
+
+    /*    The starting indices for each string*/
+    var n_edges : int;
+
+    /*    The graph is directed (True) or undirected (False)*/
+    var directed=1 : int;
```

Comment:
Yes, we can use a bool. 
At first I used the bool type. But the bool expression in Python and Chapel causes some problems in the code (maybe I used them in the wrong way), so I just use an integer to avoid the bool type problem in Python and Chapel.